### DISCUSSION TITLE
In query params, if a string value contains a ";" then the later part of the string (i.e, after ";") is ignored

### DISCUSSION THREAD

**Discussion by pralaykdas** [fastapi/fastapi#7490] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def read_root():
    return {"Hello": "World"}
```

### Description

* Open the browser and call the endpoint `/`.
* It returns a JSON with `{"Hello": "World"}`.
* But I expected it to return `{"Hello": "Sara"}`.

### Environment

* OS: [e.g. Linux / Windows / macOS]:
* FastAPI Version [e.g. 0.3.0]:

To know the FastAPI version use:

```bash
python -c "import fastapi; print(fastapi.__version__)"
```

* Python version:

To know the Python version use:

```bash
python --version
```

### Additional context

@app.get("/items/")
async def read_items(q: str):
    results = {"items": [{"item_id": "Foo"}, {"item_id": "Bar"}]}
    print(q)
    return results

if I hit this API with value "test":
   API: /items/?q=test
it prints: test

if I hit this API with value "test;test1;test2":
   API: /items/?q=test
it prints: test
"test1;test2" is ignored.

**Comment by obataku** [comment, marked as answer]
not fastapi specific; starlette [internally uses](https://github.com/encode/starlette/blob/master/starlette/datastructures.py#L400-L405) a [function called `parse_qsl` from `urllib.parse`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qsl) to parse query strings:
```python
        if isinstance(value, str):
            super().__init__(parse_qsl(value, keep_blank_values=True), **kwargs)
        elif isinstance(value, bytes):
            super().__init__(
                parse_qsl(value.decode("latin-1"), keep_blank_values=True), **kwargs
            )
```
it treats `;` like `&` as a query param separator, which is [common practice](https://en.wikipedia.org/wiki/Query_string#Web_forms), thus `test1` and `test2` are parsed as distinct params with empty values; you can confirm as much by inspecting the starlette `Request` in your endpoint:
```python
from starlette.requests import Request

@app.get("/items/")
async def read_items(req: Request, q: str):
  results = {"items": [{"item_id": "Foo"}, {"item_id": "Bar"}]}
  print(f"{q = }")
  print(f"{req.query_params.multi_items() = })
  return results
```
____
in the future please ensure your bug reports make proper use of the template format

**Comment by pralaykdas** [comment]
Thanks obataku..
That really helps.

**Comment by tiangolo** [comment]
Thanks for the help here @obataku ! :clap: :bow: 

Thanks for reporting back and closing the issue @dpralay07 :+1:
